src/Interpreters/InterpreterExplainQuery.cpp

```diff
@@ -366,14 +369,24 @@ QueryPipeline InterpreterExplainQuery::executeImpl()
     SelectQueryOptions options;
     options.setExplain();
 
+    ContextPtr query_context = getContext();
+
+    std::optional<bool> query_plan_use_new_logical_join_step = query_context->getSettingsRef()[Setting::query_plan_use_new_logical_join_step].get();
```

Comment:
We discussed that it is not expected when execution and explanation has different defaults.